index.js

```diff
@@ -31,13 +31,78 @@ SOFTWARE.
 var parseXML     = require('xml2js').parseString
   , url          = require('url')
   , EventEmitter = require('events').EventEmitter
+  , HttpProxyAgent = require('http-proxy-agent')
+  , HttpsProxyAgent = require('https-proxy-agent')
   ;
 
 // These numbers were obtained by measuring and averaging both using this module and the official speedtest.net
 var speedTestDownloadCorrectionFactor = 1.135
   , speedTestUploadCorrectionFactor   = 1.139
+  , proxyOptions = null
   ;
 
+// Search a proxy (http|https) in env with case-insensitive
+var proxyHttpEnv = findPropertiesInEnvInsensitive("HTTP_PROXY");
```

Comment:
Please add these variables to the block above

Reply:
done